File: src/com/uniplore/solution/BinarySearch.java
package com.uniplore.solution;

public class BinarySearch {

	public static boolean binarySearch(int[] arr,int start,int end , int target){
		InsertSort.insertSort(arr);  // 将数组排序
		if (start == end && target != arr[start]) {
			System.out.println("找不到对应的元素");
			return false;
		}
		
		int mid = (start + end) / 2;  //找到数组的中间位置
		if (target == arr[mid]) {
			System.out.println("找到对应的元素");
			return true;
		}else if(target > arr[mid]) {
			binarySearch(arr, mid+1, end, target);
		}else if(target < arr[mid]){
			binarySearch(arr, 0, mid-1, target);
		}
		return false;
	}
	
	public static void main(String[] args) {
		int[] arr = {-1,2,100,-100,30,-23,90};
		//binarySearch(arr, 0, arr.length-1, -10);   //测试组1
		//binarySearch(arr, 0, arr.length-1, 1000);  //测试组2
		binarySearch(arr, 0, arr.length-1, 10000);     //测试组3
	}
}

File: CommonAlgorithms/binarySearch.py
#<Python代码：二分查找。对于排好序的序列，查找数列中是否存在某个数，如果存在，则返回该数所在的位置，如果不存在，返回-1000>
def binarySearch(arr, start, end, target):
    if start >= end  and target != arr[start]:
        result = -1000     #表示找不到对应的位置，我想用-1，但是Python中-1代表数列的最后一个位置
        return result
    middlePosition = (start+end)//2   #找到中间位置
    if target == arr[middlePosition]:
        result = middlePosition+1
        return result
    elif target > arr[middlePosition]:
        result = binarySearch(arr, middlePosition+1,end,target)
    elif target < arr[middlePosition]:
        result = binarySearch(arr,start,middlePosition-1,target)
    return result
arr = [-100,-23,-1,2,30,90,100]
result  = binarySearch(arr,0,len(arr)-1,100)
print("二分查找的结果为:",result)
print("\n")
